components/aws/sagemaker/test/integration/component_tests/test_train.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+import os
+import utils
+from utils import kfp_client_utils
+from utils import argo_utils
+from utils import minio_utils
+from utils import sagemaker_utils
+
+
+def test_simple_mnist_training(
+    kfp_client, experiment_id, s3_client, sagemaker_client, region
+):
+
+    pipeline_name = "simple_mnist_training"
+    test_file_dir = os.path.join("resources/", pipeline_name)
+    run_id, status = kfp_client_utils.compile_run_monitor_pipeline(
+        kfp_client, experiment_id, test_file_dir, pipeline_name, 1200
+    )
+
+    workflow_json = kfp_client_utils.get_workflow_json(kfp_client, run_id)
+
+    if not status:
+        argo_utils.print_workflow_logs(workflow_json["metadata"]["name"])
+        pytest.fail(f"Test Failed: {pipeline_name}")
+    else:
+        print("run suceeded")
+        outputs = {"sagemaker-training-job": ["job_name", "model_artifact_url"]}
+        output_files = minio_utils.artifact_download_iterator(
+            workflow_json, outputs, test_file_dir
+        )
+
+        # Verify Training job was successful on Sagemaker
+        training_job_name = utils.extract_information(
+            output_files["sagemaker-training-job"]["job_name"], "job_name.txt"
+        )
+        print("training job name: ", training_job_name)
+        train_response = sagemaker_utils.describe_training_job(
+            sagemaker_client, training_job_name.decode()
+        )
+        assert train_response["TrainingJobStatus"] == "Completed"
+
+        # Verify model artifacts output
+        model_artifact_url = utils.extract_information(
+            output_files["sagemaker-training-job"]["model_artifact_url"],
+            "model_artifact_url.txt",
+        )
+        print("model_artifact_url: ", model_artifact_url)
+        assert (
+            model_artifact_url.decode()
```

Comment:
Can you add a comment here what exactly are you testing here? 
`train_response` directly comes from SageMaker how about `model_artifact_url` why is it from minio ?

Reply:
+1. Also are you sure that the results of a previous run won't lead to a false positive in any edge case - basically, are we ensuring that information is extracted for the correct/latest `sagemaker-training-job`?

Reply:
@mbaijal you are right, here we are checking that model artifact url returned by the component is output from the Sagemaker job that was run for the test. The training job name is generated randomly like `TrainingJob-20200429200820-AI79` and the model URL will have training job name in path. I will add that check explicitly in the next iteration and also add this comments as per gautam's suggestion. Example for full S3 path:

`s3://<bucket_name>/mnist_kmeans_example/output/TrainingJob-20200429200820-AI79/output/model.tar.gz`